File: FaceSample/app/src/main/java/com/regula/facesample/items/other/NetworkInterceptorItem.java
package com.regula.facesample.items.other;

import android.content.Context;

import com.regula.facesample.items.CategoryItem;
import com.regula.facesdk.FaceSDK;

/**
 * Created by Sergey Yakimchik on 23.08.21.
 * Copyright (c) 2021 Regula. All rights reserved.
 */

public class NetworkInterceptorItem extends CategoryItem {

    @Override
    public void onItemSelected(Context context) {
        FaceSDK.Instance().setNetworkInterceptorListener(httpURLConnection -> {
            httpURLConnection.getConnection().setRequestProperty("customFiels", "TestValue");
        });
        FaceSDK.Instance().setServiceUrl(FaceSDK.Instance().getServiceUrl());
        FaceSDK.Instance().startLiveness(context, livenessResponse -> {});
    }

    @Override
    public String getTitle() {
        return "Network interceptor";
    }

    @Override
    public String getDescription() {
        return "Adds custom http fields to the outgoing requests.";
    }
}

File: BasicSample/app/src/main/java/com/regula/facesamplekotlin/items/other/NetworkInterceptorItem.kt
package com.regula.facesamplekotlin.items.other

import android.content.Context
import com.regula.common.http.HttpRequestBuilder
import com.regula.facesamplekotlin.category.CategoryItem
import com.regula.facesdk.FaceSDK

class NetworkInterceptorItem : CategoryItem() {

    override fun onItemSelected(context: Context) {
        FaceSDK.Instance().setNetworkInterceptorListener { httpRequestBuilder: HttpRequestBuilder ->
            httpRequestBuilder.connection.setRequestProperty("fields", "value")
        }
        FaceSDK.Instance().serviceUrl = FaceSDK.Instance().serviceUrl
        FaceSDK.Instance().startLiveness(context) { }
    }

    override val title: String
        get() =  "Network interceptor"

    override val description: String
        get() = "Adds custom http fields to the outgoing requests"
}